src/praisonai-agents/tests/unit/context/test_context_compression.py

```
@@ -0,0 +1,172 @@
+        print(f"Tokens saved: {result.tokens_saved}")
+        print(f"Compression ratio: {result.compression_ratio:.2f}")
+        
+        # Verify compression worked
+        assert len(result.messages) <= len(messages)
+        assert result.final_tokens <= original_tokens
+        assert result.tokens_saved >= 0
+        
+        print("✓ ContextCompressor test passed!")
+        return True
+        
+    except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)